src/python/pants/backend/python/rules/pex.py

```diff
@@ -45,32 +51,36 @@ def create_from_adaptors(
                 for py_req in maybe_python_req_lib.requirements:
                     all_target_requirements.add(str(py_req.requirement))
         all_target_requirements.update(additional_requirements)
-        return PexRequirements(requirements=tuple(sorted(all_target_requirements)))
+        return PexRequirements(all_target_requirements)
 
 
-@dataclass(frozen=True)
+@frozen_after_init
+@dataclass(unsafe_hash=True)
 class PexInterpreterConstraints:
-    constraint_set: Tuple[str, ...] = ()
+    constraints: FrozenOrderedSet[str]
+
+    def __init__(self, constraints: Optional[Iterable[str]] = None) -> None:
+        self.constraints = FrozenOrderedSet(sorted(constraints or ()))
 
     def generate_pex_arg_list(self) -> List[str]:
         args = []
-        for constraint in sorted(self.constraint_set):
+        for constraint in sorted(self.constraints):
             args.extend(["--interpreter-constraint", constraint])
         return args
 
     @classmethod
     def create_from_adaptors(
         cls, adaptors: Iterable[TargetAdaptor], python_setup: PythonSetup
     ) -> "PexInterpreterConstraints":
-        interpreter_constraints = {
+        constraints = {
             constraint
             for target_adaptor in adaptors
             for constraint in python_setup.compatibility_or_constraints(
                 getattr(target_adaptor, "compatibility", None)
             )
             if isinstance(target_adaptor, PythonTargetAdaptor)
         }
-        return PexInterpreterConstraints(constraint_set=tuple(sorted(interpreter_constraints)))
+        return PexInterpreterConstraints(constraints)
```

Comment:
all these named arguments going away feels orthogonal to this change. I kind of like them going away because the constructors only take one argument, but still it's a bit odd.

Not a problem at all though :)